frontend/packages/dev-console/src/components/import/image-search/ImageSearch.tsx

```diff
@@ -86,6 +92,12 @@ const ImageSearch: React.FC = () => {
     !dirty && values.searchTerm && handleSearch(values.searchTerm);
   }, [dirty, handleSearch, values.searchTerm]);
 
+  React.useEffect(() => {
+    if (values.searchTerm !== initialValues.searchTerm) {
+      initialValues.route.targetPort = '';
+    }
+  }, [values.searchTerm, initialValues.searchTerm, initialValues.route.targetPort]);
```

Comment:
You've got 2 form fields now resetting the value of `initialValues.route.targetPort` creating a race condition.

My suggestion would be to track first load vs subsequent loads. The value can be stored in a ref:
`const canUpdateTargetPort = React.useRef(!initialValues.route.targetPort)`

You'd need to do this in each form field that uses this value.

Then once first load is complete do `canUpdateTargetPort.current = true`.

Reply:
So in this file `initialValues.route.targetPort` is getting re-initialized only once in line 97 whenever we have a new `searchTerm`. No other field resets its value, everywhere else `values.route.targetPort` is set and that does not affect `initialValues.route.targetPort`. Did you find any use case  where you saw the race condition? I might be missing something.

Reply:
My mistake. I mistook the other file as being in the same flow. But it's for the 2nd radio option so it wouldn't be an issue.

Did you consider just checking if the target port is found in the set of ports? And only if it isn't use the first available port? Would there be an issue with doing this? That way we don't have to mutate the initial value.

The side effect here is if they change the target port for one image search, then search again and that same target port is available, but not first in the list; it would keep that selection. I think that's ok.

Reply:
So I went through a few examples where it was needed to change the initial values and hence we have `enableReinitialize` which when set to `true` lets us change the initial values. Anyways I'll go through the other approaches you suggested. 

> Another thought is to check if the field is touched or not and if not use the initial value. Immediately after checking you can force set the field touched so that we don't use the initial value ever again.

Will try this one and see if it works.

Reply:
@divyanshiGupta This combined with my suggestion above should work as a workaround here - 
```js
  React.useEffect(() => {
    if (touched.searchTerm && initialValues.searchTerm !== values.searchTerm) {
      const targetPort: ContainerPort = _.head(values.isi.ports);
      targetPort && setFieldValue('route.targetPort', makePortName(targetPort));
    }
  }, [
    touched.searchTerm,
    setFieldValue,
    values.isi.ports,
    initialValues.searchTerm,
    values.searchTerm,
  ]);
```
The only issue I found with this approach was that if a user starts to update the searchTerm after searching for an image before, the target port resets to head. I can't think of a scenario where this would cause any real problems for user though. If a user is changing the searchTerm, he intents to update the container image which should reset the ports.